### ISSUE TITLE
_data attribute in the GiraffeViewEngine

### ISSUE THREAD

**Issue by joahim** [giraffe-fsharp/Giraffe#324]
The current implementation of the `_data` attribute in the GiraffeViewEngine does not work alongside it's specification:

https://www.w3.org/TR/html5/dom.html#embedding-custom-non-visible-data-with-the-data-attributes

In other words, it allows creation of `data="value"` but not `data-something="value"`.

This is how Fable.React defines the data attribute:

https://github.com/fable-compiler/fable-react/blob/0514b49c9459989c37db763bb01af434dfe9df9d/src/Fable.React/Fable.Helpers.React.fs#L734

It may also be nice to have an "escape route" for cases not currently covered by the DSL by providing something alongside the Fable.react's `HTMLAttr.Custom` attribute.

**Comment by dustinmoris** [comment]
> In other words, it allows creation of data="value" but not data-something="value".

Yes this is an oversight and needs to be fixed. Something similar to Fable makes sense (e.g. `_data "key" "value"`).

> It may also be nice to have an "escape route" for cases not currently covered by the DSL by providing something alongside the Fable.react's HTMLAttr.Custom attribute.

If I understand you correctly then I believe you're after [`XmlAttribute`](https://github.com/giraffe-fsharp/Giraffe/blob/develop/src/Giraffe/GiraffeViewEngine.fs#L30-L51).

**Comment by alexpantyukhin** [comment]
Hi! It looks like  `_dataCustom "key" "value"` in my [PR](https://github.com/giraffe-fsharp/Giraffe/pull/330).

**Comment by dustinmoris** [comment]
Hey, why did you create `_dataCustom` instead of fixing `_data`?

**Comment by alexpantyukhin** [comment]
Yes, you are right. Didn't correcly get that the `data` is not allowed.

**Comment by dustinmoris** [comment]
Ok no worries. I've taken your implementation and applied it to `_data`. I will release it with the next version.

**Comment by alexpantyukhin** [comment]
Then I guess using `_data` there should be fixed also: https://github.com/giraffe-fsharp/Giraffe/blob/master/tests/Giraffe.Benchmarks/Program.fs#L15 .
